actions/campaigns.go

```diff
@@ -611,3 +613,98 @@ func DeleteCampaignSchedule(c *gin.Context) {
 	})
 
 }
+
+func PatchCampaignSchedule(c *gin.Context) {
+
+	u := middleware.GetUser(c)
+
+	if !u.Boundaries.ScheduleCampaignsEnabled {
+		c.JSON(http.StatusForbidden, gin.H{
+			"message": "You do not have permission to schedule campaign, please upgrade to a bigger plan or contact support.",
+		})
+	}
+
+	campaignID, err := strconv.ParseInt(c.Param("id"), 10, 64)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"message": "Id must be an integer",
+		})
+		return
+	}
+
+	campaign, err := storage.GetCampaign(c, campaignID, u.ID)
+	if err != nil {
+		logrus.Println(err)
+		c.JSON(http.StatusNotFound, gin.H{
+			"message": "Campaign not found",
+		})
+		return
+	}
+
+	body := &params.CampaignSchedule{}
+	if err := c.ShouldBind(body); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"message": "Invalid parameters, please try again",
+		})
+		return
+	}
+
+	if err := validator.Validate(body); err != nil {
+		c.JSON(http.StatusBadRequest, err)
+		return
+	}
+
+	schAt, err := time.Parse("2006-01-02 15:04:05", body.ScheduledAt)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"message": "Invalid parameters, scheduled_at should be format: 2006-02-01 15:04:05",
+		})
+		return
+	}
+
+	// if schedule exist update.
+	if campaign.Schedule != nil {
+		campaign.Schedule.ScheduledAt = schAt
+		err = storage.CreateCampaignSchedule(c, campaign.Schedule)
```

Comment:
No need to have this block of code twice, you can just set the campaign.Schedule to a new schedule object in the `else` condition and then invoke CreateCampaignSchedule.

```
} else {
campaign.Schedule = &entities.CampaignSchedule{
		ID:          ksuid.New(),
		CampaignID:  campaign.ID,
		ScheduledAt: schAt,
	}
}
```